inference-engine/src/gna_plugin/gna_device.cpp

```diff
@@ -80,6 +80,7 @@ uint32_t GNADeviceHelper::propagate(const uint32_t requestConfigId, Gna2Accelera
     checkGna2Status(status1);
     const auto status2 = Gna2RequestEnqueue(requestConfigId, &reqId);
     checkGna2Status(status2);
+    scheduledReqIds.push_back(reqId);
```

Comment:
What is the reason to use std::vector if we are using only the last element anyway?

Reply:
Initially I've iterated over all elements, but then I assumed that it is in order queue, hence checking only last  element, there is one advantage of vector over variable which is we can check if it is empty or not. So there are 3 possible solutions:
1) keep variable of last ReqId  and boolean checking if there were requests at all
2) keep valiable of last ReqId but initialize it with some preset value
3) keep vector to keep track of all request ids which size will act as a boolean.

I do not like solution 1) because it is keeping 2 varables, and do not like solution 2 since I cannot assure that reqId generated by library will never be the preset value, even though that chance is close to 0. While solution 3 keeps some redundant data, but it can possibly be used in debugging some time in the future, because it at least give us some knowledge about enqueues made in plugin.

Please decide what strategy should be used

Reply:
I would consider solution 3 as unacceptable because typical usage scenarios might include several hours of work, so such growing vector will be definitely interpreted by customers as a memory leak.
What is the problem to keep 2 variables? :)

Reply:
Just personal need to keep classes as small as possible. I will proceed with case 1 then